### ISSUE TITLE
Enhance parsing of format for ExternalGraphic (IconSymbolizer)

### ISSUE THREAD

**Issue by dsuren1** [geostyler/geostyler#1453]
# Feature Request

With respect to IconSymbolizer, the format for ExternalGraphic is obtained from the [image](https://github.com/geostyler/geostyler-sld-parser/blob/master/src/SldStyleParser.ts#L1876), but there are some cases where the image url doesn't explicitly mentions the format (ex. https://master.demo.geonode.org/documents/1623/link). In this case even though the source is valid, the format is not obtained by the parser which poses a limitation.

**Describe the solution you'd like**
The IconSymbolizer should also allow user to specify the format or the parser should try to obtain the **content-type** of the image provided to obtain the format if missing.

**Describe alternatives you've considered**
-  There could be a support to have a `format` property for IconSymbolizer in geostyler-style which would allow the user to specify one (Limits the guessing of the format) 
```
{
  name: 'External Graphic',
  rules: [{
    name: '',
    symbolizers: [{
      kind: 'Icon',
      image: 'https://sample.test.com/testlogo',
      format: 'image/png'
      size: 10,
      rotate: 90
    }]
  }]
};
```

- Or alternatively, without modifying geostyler-style, other parsers and component, the image format could be obtained from the **content-type** when the image URL doesn't have format mentioned

**Comment by jansule** [comment]
Good point!

I would prefer to extend the geostyler-style IconSymbolizer directly.

@dsuren1 it would be awesome if you can create a PR for this

**Comment by KaiVolland** [comment]
Fixed via geostyler/geostyler-style#280
